src/wp-includes/option.php

```diff
@@ -1106,6 +1113,71 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ * Determine the appropriate autoload value based on input.
+ *
+ * This function checks the provided autoload value and returns a standardized
+ * value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * @since 6.4.0
+ *
+ * @param mixed $autoload The autoload value to evaluate.
+ *
+ * @return string The standardized autoload value: 'yes', 'no', 'default-yes', or 'default-no'.
+ */
+function get_autoload_value( $autoload ) {
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+		default:
+			return 'default-yes';
+	}
+}
+
+/**
+ * Check the size of an option value and compare it to the maximum allowed size.
+ *
+ * This function serializes the given value and checks its size against the maximum allowed
+ * option size defined by the 'max_option_size' filter.
+ *
+ * @since 6.4.0
+ *
+ * @param mixed  $value The value of the option to be checked.
+ * @param string $name  The name of the option.
+ *
+ * @return string Returns 'default-no' if the size exceeds the maximum allowed size, or 'default-yes' otherwise.
+ */
+function check_option_size( $value, $name ) {
```

Comment:
We need to pass `$name` here for context for the filter. I think this context is required to get this committed.

Reply:
Makes sense. Though let's change the parameter name to `$option` and reverse the order like is used elsewhere.
```suggestion
function check_option_size( $option, $value ) {
```

Please keep in mind to then update the parameter docs accordingly as well.